posit-dev/positron
# remove python extension override of PATH variable

Related to #4511 

the Python extension is using a newer version of VSCode, with a `GlobalEnvironmentVariableCollection` which overwrites other extension's contributions to environment variables, such as PATH. This moves to using `EnvironmentVariableCollection`, which will only touch its extension contributions.

From what I understand, the original change was made to be able to set multiple folder level scopes in a workspace, eg, set different interpreters for different folders inside a workspace: https://github.com/microsoft/vscode-python/issues/20492#issuecomment-1379454451 and https://github.com/microsoft/vscode/issues/171173
Positron operates (or at least the interpreter dropdown UI shows) 1 interpreter per workspace, so I don’t think the scoping stuff applies to us.

### QA Notes

- make sure Quarto is in your local `positron` directory. AFAICT this does not automatically populate on `yarn` runs, so you might have to put it there manually. (At build time, it is at `~/Applications/Positron.app/Contents/Resources/app/quarto/bin`, which is being properly identified by the Quarto extension 👍)
- delete other local Quarto installations (likely by running `rm -fr ~/Applications/quarto`)
- open up Positron
- run `quarto --version` in terminal, should not give any errors
